api/grpcserver.go

```diff
@@ -336,19 +336,21 @@ func (svr *GRPCServer) SendAction(ctx context.Context, in *iotexapi.SendActionRe
 	if in.GetAction().GetCore().GetChainID() > 0 {
 		_apiCallSourceWithChainIDMtc.WithLabelValues(chainID).Inc()
 	} else {
-		var clientIP, sender, recipient string
-		selp, err := (&action.Deserializer{}).ActionToSealedEnvelope(in.GetAction())
+		var (
+			clientIP, recipient string
+			ok                  bool
+		)
+		selp, err := (&action.Deserializer{}).SetEvmNetworkID(svr.coreService.EVMNetworkID()).ActionToSealedEnvelope(in.GetAction())
 		if err != nil {
 			return nil, err
 		}
 		if p, ok := peer.FromContext(ctx); ok {
 			clientIP, _, _ = net.SplitHostPort(p.Addr.String())
 		}
-		if senderAddr, err := address.FromBytes(selp.SrcPubkey().Hash()); err == nil {
-			sender = senderAddr.String()
+		if recipient, ok = selp.Destination(); !ok {
+			recipient = "staking-action"
```

Comment:
staking action does not implement `Destination()` interface, so will be `""`

Reply:
ToEthTx()